java/compaction/compaction-task-creation/src/main/java/sleeper/compaction/task/creation/RunTasks.java

```diff
@@ -122,48 +126,51 @@ public void run() {
         int queueSize = QueueMessageCount.withSqsClient(sqsClient).getQueueMessageCount(sqsJobQueueUrl)
                 .getApproximateNumberOfMessages();
         LOGGER.info("Queue size is {}", queueSize);
+        // Request 1 task for each item on the queue
+        run(startTime, queueSize);
+    }
+
+    private void run(int requestedTasks) {
+        run(System.currentTimeMillis(), requestedTasks);
+    }
 
+    private void run(long startTime, int requestedTasks) {
+        if (requestedTasks == 0) {
+            LOGGER.info("Finishing as number of tasks requested was 0");
+            return;
+        }
         // Find out number of pending and running tasks
         int numRunningAndPendingTasks = CommonJobUtils.getNumPendingAndRunningTasks(clusterName, ecsClient);
         LOGGER.info("Number of running and pending tasks is {}", numRunningAndPendingTasks);
 
         int maxNumTasksToCreate = maximumRunningTasks - numRunningAndPendingTasks;
-        LOGGER.info("Maximum number of tasks to create is {}", maxNumTasksToCreate);
-
-        // Do we need to scale?
+        if (maximumRunningTasks == numRunningAndPendingTasks) {
+            LOGGER.info("Finishing as maximum running tasks of {} has been reached", maximumRunningTasks);
+            return;
+        }
```

Comment:
This if statement could handle the case where more tasks are running than the maximum. That could easily happen if someone runs the new script at the same time as the Lambda runs.